doc/developing/os-compatibility.md

```diff
@@ -0,0 +1,93 @@
+OS/library compatibility policy
+===============================
+
+This document describes how we decide which versions of operating systems, C++ standards, libraries,
+build tools (CMake) or compilers Minetest requires as a minimum.
+
+As most development happens on Linux the first measure is to check the version of the component in question on:
+* the oldest still-supported **Ubuntu** (End of Standard Support)
+* the oldest still-supported **Debian** (*not* extended LTS)
+* optional: the second newest **RHEL (derivative)**
+
+Generally this leads to versions about 5 years old and works as a reasonable result for BSDs and other platforms too.
+
+Needless to say that any such changes need to work on our other platforms too, as listed below.
+
+### Windows
+
+We currently support Windows 8 or later.
+
+Despite requiring explicit support code in numerous places there doesn't seem to be a strong case for dropping older Windows versions. We will likely only do it once SDL2 does so.
+
+Note that we're constrained by the versions [vcpkg](https://vcpkg.io/en/packages) offers, for the MSVC build.
+
+### macOS
+
+We currently support macOS 10.14 (Mojave) or later.
+
+Since we do not have any macOS developer we can only do some shallow testing in CI.
+So this is subject to change basically whenever Github throws
+[a new version](https://github.com/actions/runner-images?tab=readme-ov-file#available-images) at us, or for other reasons.
+
+### Android
+
+We currently support Android 5.0 (API 21) or later.
+
+Before raising this it's a good idea to check the Google Play Console to see what our userbase is using.
```

Comment:
Usually the minimum Android version has only been increased when the latest Android NDK drops support for older versions, which was the case with support being dropped for Jelly Bean (4.1-4.3) in r23 and KitKat (4.4) in r26. I don't think there would be any reason to drop older Android versions as long as the latest NDK supports it so the minimum version would basically be controlled by what the latest NDK supports.

Reply:
are the NDK necessarily keep droping support on old Android versions or it can still support but don't want to?